airbyte-integrations/connectors/source-google-drive/source_google_drive/stream_reader.py

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import io
+import json
+import logging
+import re
+from datetime import datetime
+from io import IOBase
+from typing import Any, Dict, Iterable, List, Optional, Set
+
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException, FailureType
+from google.oauth2 import credentials, service_account
+from googleapiclient.discovery import build
+from googleapiclient.http import MediaIoBaseDownload
+
+from .spec import SourceGoogleDriveSpec
+
+FOLDER_MIME_TYPE = "application/vnd.google-apps.folder"
+GOOGLE_DOC_MIME_TYPE = "application/vnd.google-apps.document"
+EXPORTABLE_DOCUMENTS_MIME_TYPES = [
+    GOOGLE_DOC_MIME_TYPE,
+    "application/vnd.google-apps.spreadsheet",
+    "application/vnd.google-apps.presentation",
+    "application/vnd.google-apps.drawing",
+]
```

Comment:
Good point @aaronsteers, we most certainly will want a separate parser for Excel eventually.

I can definitely see a situation where a user doesn't really care what files are in their folder they just want everything that can be parsed. But, if we wanted to offer that I think maybe it would be best done as an umbrella parser that can call out to other parsers rather than including Excel in with this one.